tools/walletextension/wallet_extension.go

```diff
@@ -346,3 +355,50 @@ func adjustStateRoot(rpcResp interface{}, respMap map[string]interface{}) {
 		}
 	}
 }
+
+// getStorageAtInterceptor checks method name in the request and if it is one of the methods that we use differently than an eth standard.
+// At the moment, it is only getStorageAt method. It returns response if we should use that response or nil otherwise
+func (w *WalletExtension) getStorageAtInterceptor(request *accountmanager.RPCRequest, hexUserID string) map[string]interface{} {
+	response := map[string]interface{}{}
+	// all responses must contain the request id. Both successful and unsuccessful.
+	response[common.JSONKeyRPCVersion] = jsonrpc.Version
+	response[common.JSONKeyID] = request.ID
+
+	// check if parameters are correct, and we can intercept a request, otherwise return nil
+	if w.checkParametersForInterceptedGetStorageAt(request.Params) {
+		// check if userID in the parameters is also in our database
+		userID, err := common.GetUserIDbyte(hexUserID)
+		if err != nil {
+			w.logger.Warn("GetStorageAt called with appropriate parameters to return userID, but not found in the database: ", "userId", hexUserID)
+			return nil
+		}
+
+		key, err := w.storage.GetUserPrivateKey(userID)
+		if err != nil || len(key) == 0 {
+			w.logger.Info("Trying to get userID, but it is not present in our database: ")
+			return nil
+		}
+
+		response[common.JSONKeyResult] = hexUserID
+		return response
+	}
+	w.logger.Info(fmt.Sprintf("parameters used in the request do not match requited parameters for interception: %s", request.Params))
+
+	return nil
+}
+
+func (w *WalletExtension) checkParametersForInterceptedGetStorageAt(params []interface{}) bool {
+	if len(params) != 3 {
+		w.logger.Info(fmt.Sprintf("getStorageAt expects 3 parameters, but %d received", len(params)))
+		return false
+	}
+
+	address, _ := params[0].(string)
+	position, _ := params[1].(string)
+
+	// we only want to intercept getStorageAt if those parameters match pre-defined ones
+	if address == common.GetStorageAtUserIDSpecialAddress && position == common.GetStorageAtUserIDSpecialPosition {
+		return true
```

Comment:
What if there was only one special field ?

```suggestion
      if methodName, ok := params[0].(string); ok {
        return methodName == common.GetStorageAtUserIDRequestMethodName
      }
      return false
```

Reply:
fixed.